pkg/plugin/aggregation/aggregator.go

```diff
@@ -217,5 +225,20 @@ func (a *Aggregator) handleResult(result *plugin.Result) error {
 	io.Copy(f, result.Body)
 	glog.Infof("wrote results to %v\n", resultsFile)
 
+	// If it's a tarball, extract it
+	if extension == ".tar.gz" {
+		f.Close()
```

Comment:
Trying to be defensive in case later code introduces things that may panic... Close() is safe to call repeatedly on a filehandle in go, at least from what I've been able to dig up.

I'll make another pass at this and see if I can make it a bit cleaner, I'm not all that happy with it yet.